### PR TITLE
[IoT] BREAKING CHANGE: `iot dps access-policy`: Deprecate access-policy in favor of policy

### PR DESCRIPTION
**Pull request by vilit1**
**Description**

Replace access-policy with policy in `az iot dps` to align better with other command groups.

**Testing Guide**
The following command was run to test:
`azdev test test_dps_lifecycle test_dps_certificate_lifecycle test_dps_linked_hub_lifecycle`

In addition, the entire iot module was tested:
`azdev test iot`

**History Notes**

The command group `az iot dps access-policy` has been deprecated and replaced with `az iot dps policy` .

Related prs:
https://github.com/Azure/azure-cli/pull/20682
https://github.com/Azure/azure-cli/pull/21892

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Review by c-ryan-k — APPROVED** [review]
LGTM - thanks for jumping on this so quickly 👍

**Comment by yonzhan** [comment]
iot

File: src/azure-cli/azure/cli/command_modules/iot/_params.py [hunk 1]
```diff
@@ -78,55 +78,22 @@ def load_arguments(self, _):  # pylint: disable=too-many-statements
...
-        c.argument('rights', options_list=['--rights', '-r'], nargs='+',
-                   arg_type=get_enum_type(AccessRightsDescription),
-                   help='Access rights for the IoT Hub Device Provisioning Service. '
-                        'Use space-separated list for multiple rights.')
-        c.argument('primary_key', help='Primary SAS key value.')
-        c.argument('secondary_key', help='Secondary SAS key value.')
-
     with self.argument_context('iot dps policy') as c:
-        c.argument('access_policy_name', options_list=['--policy-name', '--pn'],
+        c.argument('access_policy_name', options_list=['--policy-name', '--pn', '--name', '-n'],
```
Comment (digimaun) [inline src/azure-cli/azure/cli/command_modules/iot/_params.py:86]: Why add --name and -n as arg aliases for policy name? We used this deprecation opportunity to explicitly align arg aliases with the IoT extension command set where `-n` generally refers to the target parent ARM resource in question, which is a pivot from the old access-policy.

This is the presentation with 2.34.1 which is what we wanted to align on.
image

**Review by digimaun — CHANGES_REQUESTED** [review]
Align on arg alias.

File: src/azure-cli/azure/cli/command_modules/iot/_params.py (hunk 1 above)
Comment (vilit1) [reply src/azure-cli/azure/cli/command_modules/iot/_params.py]: done - now dps_name refers to --dps-name, --name, and -n

image

Comment (digimaun) [reply src/azure-cli/azure/cli/command_modules/iot/_params.py]: Thank you.

I believe we will need new test recordings for the changed tests.

**Comment by jiasli** [comment]
Deprecation version was deferred by https://github.com/Azure/azure-cli/pull/21892.

**Review by zhoxing-ms — APPROVED** [review]

**Review by digimaun — APPROVED** [review]

**Pull request merged**